test/url-to-options.js

```diff
@@ -0,0 +1,56 @@
+import url from 'url';
+import test from 'ava';
+import urlToOptions from '../source/url-to-options';
+
+const githubURL = 'https://user:password@github.com:443/';
+
+test('converts node legacy URL to options', t => {
+	const parsedURL = url.parse(githubURL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		hash: null,
+		hostname: 'github.com',
+		href: githubURL,
+		path: '/null',
+		pathname: '/',
+		port: 443,
+		protocol: 'https:',
+		search: null
+	};
+
+	t.deepEqual(options, expected);
+});
+
+test('converts URL to options', t => {
+	const parsedURL = new url.URL(githubURL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		auth: 'user:password',
+		hash: '',
+		hostname: 'github.com',
+		href: 'https://user:password@github.com/',
+		path: '/',
+		pathname: '/',
+		protocol: 'https:',
+		search: ''
+	};
+
+	t.deepEqual(options, expected);
+});
+
+test('converts IPv6 URL to options', t => {
+	const IPv6URL = 'https://[2001:cdba::3257:9652]:443/';
+	const parsedURL = new url.URL(IPv6URL);
+	const options = urlToOptions(parsedURL);
+	const expected = {
+		hash: '',
+		hostname: '2001:cdba::3257:9652',
+		href: 'https://[2001:cdba::3257:9652]/',
+		path: '/',
+		pathname: '/',
+		protocol: 'https:',
+		search: ''
+	};
```

Comment:
Shouldn't be there `expected.port = 443`?

Reply:
What can I tell you 😅 ? Computer says no.

Reply:
You could change the port to another, e.g. to `8080` or `80`. But I don't know if it's a good idea.

I would just add a comment: `// IPv6 + :443 = no port`

Reply:
The comment shouldn't be on a separate line I think :) Maybe (?):

```js
	const parsedURL = new url.URL(IPv6URL); // TODO: Use the `URL` global when targeting Node.js 10
```

Reply:
Just to be a little more clear 😅 , if Node feels that the default HTTPS port should not be added if the protocol is already HTTPS, I guess that's fine. I wouldn't write code to go against what Node does.

Reply:
That's https? Then sorry, hadn't noticed that :D